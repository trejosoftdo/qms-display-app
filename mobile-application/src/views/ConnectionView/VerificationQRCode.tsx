import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import QRCode from 'react-native-qrcode-svg';

/**
 * VerificationQRCodeProps defines the props for the Verification QR Code component.
 *
 * @property verificationURI - Verification URI
 */
interface VerificationQRCodeProps {
  verificationURI?: string;
}

/**
 * A component to display the verification QR code 
 *
 * @param {VerificationQRCodeProps} props - The props for the Verification QR Code component.
 */
const VerificationQRCode: React.FC<VerificationQRCodeProps> = (props: VerificationQRCodeProps) => (
  <View style={styles.container}>
    <QRCode value={props?.verificationURI} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    marginBottom: 32,
  },
});


export default VerificationQRCode;