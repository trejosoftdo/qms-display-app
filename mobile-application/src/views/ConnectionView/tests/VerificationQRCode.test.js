import React from 'react';
import VerificationQRCode from '../VerificationQRCode';

describe('VerificationQRCode component', () => {
  const mockProps = {
    verificationURI: 'https://verification-uri.test',
  };

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders the component as expected', () => {
    expect(renderShallow(<VerificationQRCode {...mockProps} />)).toMatchSnapshot();
  });
});
