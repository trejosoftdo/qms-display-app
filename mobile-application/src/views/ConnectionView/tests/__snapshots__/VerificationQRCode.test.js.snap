// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerificationQRCode component renders the component as expected 1`] = `
<View
  style={
    {
      "marginBottom": 32,
    }
  }
>
  <QRCode
    value="https://verification-uri.test"
  />
</View>
`;
